### PR TITLE
Feature mdt-78 regenerate lost key

### PR DESCRIPTION
**Pull request by lisatr2812**

### CODE REVIEW

File: src/apis/impl/NotificationApiImpl.ts [hunk 1]
```diff
@@ -154,6 +157,14 @@ export class NotificationApiImpl implements NotificationApi {
         )
   }
 
+  private _findAfterDateFilterValue(afterDate?: number): number {
+    if (afterDate != undefined) {
+      return afterDate
+    }
+
+    return new Date().getTime() - 1000 * 60 * 60 * 24 * 30
+  }
```
Comment (LotuxPunk) [inline src/apis/impl/NotificationApiImpl.ts:166]:
```suggestion
  private _findAfterDateFilterValue(fromDate?: number): number {
    if (fromDate != undefined) {
      return fromDate
    }

    return new Date().getTime() - 1000 * 60 * 60 * 24 * 30
  }
```

**Review by TestaDiRapa — APPROVED** [review]

File: src/apis/NotificationApi.ts [hunk 2]
```diff
@@ -41,9 +41,11 @@ export interface NotificationApi {
 
   /**
    * Gets all the Notifications with status "pending" that the current dataOwner can access
+   *
+   * @param afterDate : Default value is now less 30 days
    * @return an Array of the Notifications matching those criteria
    */
-  getPendingNotifications(): Promise<Array<Notification>>;
+  getPendingNotificationsAfter(afterDate?:number): Promise<Array<Notification>>;
```
[the remark this answers is not in the document]
Comment (lisatr2812) [reply src/apis/NotificationApi.ts]: Ok, I renamed the afterDate to fromDate for the variables

[the remark this answers is not in the document]
Comment (lisatr2812) [reply src/apis/NotificationApi.ts]: Also, for the console.log that should not be there, I replaced them with the errorHandler

**Comment by lisatr2812** [comment]
Duplication is due to test code, where it's better to duplicate the different conditions

**Pull request merged**